File: Lib/Telerik/Examples/QuickStart/Scheduler/CustomElements/MySchedulerCellElement.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using Telerik.WinControls;
using Telerik.WinControls.UI;

namespace Telerik.Examples.WinControls.Scheduler.CustomElements
{
    public class MySchedulerCellElement : SchedulerCellElement
    {
        LightVisualElement hoverElemnt;
       
        public MySchedulerCellElement(RadScheduler scheduler, SchedulerView view) : base(scheduler, view)
        {
            scheduler.ContextMenuOpening += scheduler_ContextMenuOpening;
        }

        void scheduler_ContextMenuOpening(object sender, SchedulerContextMenuOpeningEventArgs e)
        {
            if (!(e.Element is AppointmentElement))
            {
                e.Cancel = true;
            }
        }

        protected override void CreateChildElements()
        {
            base.CreateChildElements();
            hoverElemnt = new LightVisualElement();
            hoverElemnt.ShouldHandleMouseInput = true;
            hoverElemnt.NotifyParentOnMouseInput = false;
            hoverElemnt.StretchHorizontally = true;
            hoverElemnt.Text = "";
            hoverElemnt.TextAlignment = ContentAlignment.MiddleLeft;
            hoverElemnt.DrawFill = true;
            hoverElemnt.ForeColor = ColorTranslator.FromHtml("#FFB452");
            hoverElemnt.GradientStyle = GradientStyles.Solid;
          
            hoverElemnt.MouseEnter += buttonElemnt_MouseEnter;
            hoverElemnt.MouseLeave += buttonElemnt_MouseLeave;
            hoverElemnt.Click += hoverElemnt_Click;
         
            this.Children.Add(hoverElemnt);
        }

        void hoverElemnt_Click(object sender, EventArgs e)
        {
            DateTime end = this.Date + this.Duration;

            DateTimeInterval interval = new DateTimeInterval(this.Date, end);
            this.Scheduler.AddNewAppointmentWithDialog(interval, false, null);
        }

        void buttonElemnt_MouseLeave(object sender, EventArgs e)
        {
            hoverElemnt.BackColor = Color.Transparent;        
            hoverElemnt.Text = "";
        }

        void buttonElemnt_MouseEnter(object sender, EventArgs e)
        {
            hoverElemnt.BackColor = ColorTranslator.FromHtml("#FFFBD8");           
            hoverElemnt.Text = "Click here to add event";
        }
        
        protected override Type ThemeEffectiveType
        {
            get
            {
                return typeof(SchedulerCellElement);
            }
        }
    }
}

File: Lib/Telerik/Examples/QuickStart/Scheduler/CustomElements/MySchedulerCellElement.vb
Imports System.Linq
Imports System.Text
Imports Telerik.WinControls
Imports Telerik.WinControls.UI

Namespace Telerik.Examples.WinControls.Scheduler.CustomElements
	Public Class MySchedulerCellElement
		Inherits SchedulerCellElement
		Private hoverElemnt As LightVisualElement

		Public Sub New(ByVal scheduler As RadScheduler, ByVal view As SchedulerView)
			MyBase.New(scheduler, view)
			AddHandler scheduler.ContextMenuOpening, AddressOf scheduler_ContextMenuOpening
		End Sub

		Private Sub scheduler_ContextMenuOpening(ByVal sender As Object, ByVal e As SchedulerContextMenuOpeningEventArgs)
			If Not(TypeOf e.Element Is AppointmentElement) Then
				e.Cancel = True
			End If
		End Sub

		Protected Overrides Sub CreateChildElements()
			MyBase.CreateChildElements()
			hoverElemnt = New LightVisualElement()
			hoverElemnt.ShouldHandleMouseInput = True
			hoverElemnt.NotifyParentOnMouseInput = False
			hoverElemnt.StretchHorizontally = True
			hoverElemnt.Text = ""
			hoverElemnt.TextAlignment = ContentAlignment.MiddleLeft
			hoverElemnt.DrawFill = True
			hoverElemnt.ForeColor = ColorTranslator.FromHtml("#FFB452")
			hoverElemnt.GradientStyle = GradientStyles.Solid

			AddHandler hoverElemnt.MouseEnter, AddressOf buttonElemnt_MouseEnter
			AddHandler hoverElemnt.MouseLeave, AddressOf buttonElemnt_MouseLeave
			AddHandler hoverElemnt.Click, AddressOf hoverElemnt_Click

			Me.Children.Add(hoverElemnt)
		End Sub

		Private Sub hoverElemnt_Click(ByVal sender As Object, ByVal e As EventArgs)
			Dim [end] As Date = Me.Date + Me.Duration

			Dim interval As New DateTimeInterval(Me.Date, [end])
			Me.Scheduler.AddNewAppointmentWithDialog(interval, False, Nothing)
		End Sub

		Private Sub buttonElemnt_MouseLeave(ByVal sender As Object, ByVal e As EventArgs)
			hoverElemnt.BackColor = Color.Transparent
			hoverElemnt.Text = ""
		End Sub

		Private Sub buttonElemnt_MouseEnter(ByVal sender As Object, ByVal e As EventArgs)
			hoverElemnt.BackColor = ColorTranslator.FromHtml("#FFFBD8")
			hoverElemnt.Text = "Click here to add event"
		End Sub

		Protected Overrides ReadOnly Property ThemeEffectiveType() As Type
			Get
				Return GetType(SchedulerCellElement)
			End Get
		End Property
	End Class
End Namespace